### PR TITLE
Bugfix FXIOS-4974 [v108] External link is opened in incorrect mode

### PR DESCRIPTION
**Pull request by nbhasin2**
[Jira](https://mozilla-hub.atlassian.net/browse/FXIOS-4974) | [Github Issue](https://github.com/mozilla-mobile/firefox-ios/issues/11982)

### CODE REVIEW

File: Client/Application/NavigationRouter.swift [hunk 1]
```diff
@@ -140,12 +140,20 @@ enum NavigationPath {
...
-        case .glean(let url): NavigationPath.handleGlean(url: url)
-        case .closePrivateTabs: NavigationPath.handleClosePrivateTabs(with: bvc)
+        case .fxa(let params):
+            NavigationPath.handleFxA(params: params, with: bvc)
+        case .deepLink(let link):
+            NavigationPath.handleDeepLink(link, with: bvc)
+        case .url(let url, _):
+            // Note: The we get browser private mode from current selected tab
+            let isPrivate = bvc.tabManager.selectedTab?.isPrivate ?? false
+            NavigationPath.handleURL(url: url, isPrivate: isPrivate, with: bvc)
```
Comment (nbhasin2) [inline Client/Application/NavigationRouter.swift:150]: This is the only change required however I modified other areas to match the style

**Comment by nbhasin2** [comment]
Not particularly tagging anyone as I'd like to run a few tests before merging

File: Client/Application/NavigationRouter.swift (hunk 1 above)
Comment (nishant2718) [reply Client/Application/NavigationRouter.swift:150]: Should we update the userDefaults key for `wasLastSessionPrivate` here, or remove it altogether from the url case?

Comment (nbhasin2) [reply Client/Application/NavigationRouter.swift:150]: No, I was actually thinking of using this but update the proper state. Going to push another commit if my idea works 🤞🏼

**Review by nishant2718 — APPROVED** [review]

**Comment by nbhasin2** [comment]
image

**Pull request merged**